src/main/java/io/vertx/ext/mail/impl/SMTPConnection.java

```diff
@@ -86,6 +99,18 @@ void shutdown() {
     }
   }
 
+  void writeCommands(List<String> commands, Handler<List<String>> resultHandler) {
+    String cmds = String.join("\r\n", commands);
+    this.nsHandler.setExpected(commands.size());
+    this.write(cmds, r -> {
+      try {
+        resultHandler.handle(Arrays.asList(r.split("\r\n")));
```

Comment:
String#split(regex) should be avoided and instead rather compile the \r\n pattern and split around it using a loop (can be an util)

Reply:
Done by introducing a static final compiled `\r\n` pattern.